src/client/log/log.js

```diff
@@ -59,8 +60,11 @@ export class GameLog extends React.Component {
         currentTurn = [];
       } else {
         const args = item.move.args || [];
+        const playerID = item.move.playerID;
+        const playerNumber = playerIDs.has(playerID) ? [...playerIDs].indexOf(playerID) : playerIDs.add(playerID).size - 1;
```

Comment:
Change playerIDs to a map from playerID to playerNumber so you don't have to do this transformation here?

You can do something like:
```
if not in map:
  map[playerID] = map.size()
```